### DISCUSSION TITLE
mqtt broker

### DISCUSSION THREAD

**Discussion by woupi** [camueller/SmartApplianceEnabler#423] [Installation und Betrieb]
Leider muss ich doch nochmal Frage, bekomme es offensichtlich nicht hin.

Ich habe in einem lxc auf Proxmox seit ca einem Jahr SAE ohne Probleme am laufen. Zur Zeit Version 1.6.18.

Nun wollte ich auf die 2.1.0 umsteigen und habe dazu die Anleitungen gelesen.
Zuerst habe ich einen Klon des lxc erstellt, dadurch kommt an den schonmal nichts dran.
Dann habe ich auf Ubuntu 22.04.1 LTS upgedatet.

Anschließend habe ich den MQTT-Broker mosquitto wie unter https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/InstallationManual_DE.md#mqtt-broker
beschrieben, installiert.

Weiter steht auf der update Seite https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Update_DE.md :
`Wenn der MQTT-Broker nicht über localhost:1883 erreichbar ist oder Benutzername/Passwort notwendig sind, müssen diese Parameter in den Einstellungen des Smart Appliance Enabler konfiguriert werden.`

Wenn ich nun im Browser localhost:1883 eingebe, kommt "Seite nicht Erreichbar".
SAE läuft und wenn ich den mosquitto stoppe, wird SAE nicht mehr richtig angezeigt, darum war ich davon ausgegangen, dass alles richtig ist, ist es aber offensichtlich nicht, da keine Daten mehr an SMA gehen.

Auf einer VM läuft auch noch iobroker und da ist ein mqtt Broker Adapter installiert, kann es damit zusammen hängen, oder muss ich noch was konfigurieren?

VG

**Comment by camueller** [comment]
> Wenn ich nun im Browser localhost:1883 eingebe, kommt "Seite nicht Erreichbar".

MQTT ist nicht HTTP - also ist es normal, dass der Browser nichts anzeigt.

> AE läuft und wenn ich den mosquitto stoppe, wird SAE nicht mehr richtig angezeigt, darum war ich davon ausgegangen, dass alles richtig ist, ist es aber offensichtlich nicht, da keine Daten mehr an SMA gehen.

Die Web-Oberfläche des SAE funktioniert auch ohne MQTT. Wenn dieses also nicht richtig angezeigt wird, stimmt etwas mit dem Setup bzw. der Installation nicht. Details kann ich Dir mangels Kenntnis Deiner Virtualisierungssoftware nicht nennen.

**Comment by woupi** [comment]
> wird SAE nicht mehr richtig angezeigt, 

War eventuell nicht klar ausgedrückt.

Auf der Status Seite kommt ein roter Balken in dem steht, 
`Aktuell ist kein MQTT-Broker konfiguriert oder dieser ist nicht erreichbar. Solange dieses Problem nicht behoben ist, kann der Smart Appliance Enabler nicht funktionieren.`

Ich kann den vom Iobroker konfigurieren (IP Adresse im SAE unter MQTT eingeben), dann wird mir die Statusseite angezeigt, Daten kommen im semp allerdings trotzdem nicht.

**Comment by camueller** [comment]
> Auf der Status Seite kommt ein roter Balken in dem steht,
Aktuell ist kein MQTT-Broker konfiguriert oder dieser ist nicht erreichbar. Solange dieses Problem nicht behoben ist, kann der Smart Appliance Enabler nicht funktionieren.

Diese Meldung kommt nur nach der Konfiguration eines MQTT-Brokers im SAE bis zum Restart des SAE. Ist ein Anzeige-Fehler im SAE und für die nächste Version behoben. Für die  Funktionisweise des SAE ist entscheidend, dass im SAE-Log keine Fehler geloggt werden beim Versenden von MQTT-Nachrichten.

**Reply by woupi** [reply]
ok, komisch.
Stoppe ich mosquitto, ist der Balken da, starte ich mosquitto, ist der Balken weg.
Es sieht für mich genauso aus, als wäre das richtig, da ja kein mqtt Broker läuft bzw konfiguriert ist

**Comment by woupi** [comment]
Irgendwie habe ich offensichtlich ein Problem mit Mqtt.
Ich habe jetzt mehrfach neu gestartet und mit `sudo systemctl status mosquitto` den Status überprüft. 

`* mosquitto.service - Mosquitto MQTT Broker
     Loaded: loaded (/lib/systemd/system/mosquitto.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-12-30 17:18:17 UTC; 12h ago
       Docs: man:mosquitto.conf(5)
             man:mosquitto(8)
    Process: 115 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, status=0/>
    Process: 118 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, status=0/>
    Process: 123 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=0/SUCC>
    Process: 130 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=0/SUCC>
   Main PID: 131 (mosquitto)
      Tasks: 1 (limit: 18799)
     Memory: 3.7M
        CPU: 9.172s
     CGroup: /system.slice/mosquitto.service
             -131 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf
Dec 30 17:18:17 SAE systemd[1]: Starting Mosquitto MQTT Broker...
Dec 30 17:18:17 SAE mosquitto[131]: 1672420697: Loading config file /etc/mosquitto/conf.d/sma>
Dec 30 17:18:17 SAE systemd[1]: Started Mosquitto MQTT Broker.`


Läuft also.

Auf der web Oberfläche vom SAE ist der rote Balken nicht zu sehen und ich kann auch auf den ersten Blick alles machen.
Aber im SEMP werden keine Daten angezeigt.

Umgeschaltet auf den lxc mit 1.6.18 läuft wieder alles.

Nochmal kurz was ich gemacht habe.
- Klon vom lxc SAE erstellt
- auf den Klon mosquitto installiert
- die neue `SmartApplianceEnabler-*.war` ins Verzeichnis `/opt/sae` kopiert
- anschließend neugestartet mit `sudo systemctl restart smartapplianceenabler.service`

Fällt irgendetwas auf, was ich vergessen oder falsch gemacht haben könnte?

**Comment by camueller** [comment]
> Auf der web Oberfläche vom SAE ist der rote Balken nicht zu sehen und ich kann auch auf den ersten Blick alles machen.

Bitte meinen gestrigen Post beachten: Roter Balken irrelevant, Logs sind relvant.

Wenn Du willst, kannst diese modifizierte 2.1.0-Version verwenden, bei der die fälschliche Anzeige der roten Nachricht behoben ist: https://drive.google.com/file/d/1rPzlrsDuQ7b5daknky_FwXm6N6Ham0xQ/view?usp=share_link

Ansonsten ist das ein Problem der Komminikation zwischen Containern (SAE und Mosquitto), bei dem ich leider nicht weiterhelfen kann.

**Comment by woupi** [comment]
Der rote Balken stört mich nicht, war nur zur Klarstellung und wie ich gestern geschrieben habe, macht der komischerweise bei mir Sinn, da er nur da ist, wenn auch kein Mqtt Broker gestartet oder konfiguriert ist.

> Ansonsten ist das ein Problem der Komminikation zwischen Containern (SAE und Mosquitto), bei dem ich leider nicht weiterhelfen kann.

Ok, trotzdem Danke, eventuell schreibt aber ja ein andere etwas, sonst, würde ich es tatsächlich auch nochmal im Photovoltaikforum oder iobroker, da sind ja jede Menge user die sich mit Mqtt auskennen, versuchen.

Ich persönlich habe gar keine Ahnung, nicht mal wo ich anfangen sollte und ja Du hast geschrieben in den logs, nur da weis ich gerade nicht mal mehr wie ich daran komme (NUC, Proxmox, SAE lxc) und selbst wenn ich an die logs gekommen bin, kann ich da auch nichts mit anfangen.

Aber ich suche weiter und wenn ich es gar nicht hin bekomme, lass ich einfach die 1.6.18 weiter laufen, die macht ja bei mir das was sie soll und das jetzt schon lange und zuverlässig.

Und wie immer, vielen Dank für deine Arbeit, mehr und natürlich die kleine Spende die ich geleistet habe, kann ich nicht machen, das ist aber das mindeste.

VG

**Comment by woupi** [comment]
Ich habe jetzt das gleiche Spielchen nochmal komplett von vorne begonnen und als Update Datei, deine mir dankenswerterweise zur Verfügung gestellten neuen Datei genommen. 

Das läuft und es werden auch Daten übertragen.

Wo jetzt der Fehler gelegen hat, weiß ich natürlich nicht, aber ist eigentlich auch egal, also zu mindestens mir.

Wenn Du das aber noch irgend wie wissen möchtest, dann kann ich mich natürlich mit dem anderen lxc und den log Files noch auseinandersetzen, wie ich an die log Fils komme weiß ich jetzt wieder (Fillezilla). Wenn da Monate oder in dem Fall ein Jahr zwischen ist, muss ich leider immer wieder von vorne beginnen, ich glaube und hoffe, dass ist meinem alter geschuldet, oder weil ich zu viel andere Baustellen habe, keine Ahnung.

Im update Prozess habe ich nichts anders gemacht, meine ich zumindest, ausschließen will ich das nicht, da keiner unfehlbar ist und das behaupte ich auch nicht von mir, bin mir aber sehr sicher, da ich jeden Schritt nach deiner Anleitung gemacht habe.

Danke nochmal und guten Rutsch ins neue Jahr

**Comment by woupi** [comment]
Mir ist gerade doch noch ein Unterschied eingefallen:
Node-Red

Bei dem ersten Update Versuch, habe ich mit Node-Red, wie in deiner Anleitung beschrieben, weiter gemacht.

Beim 2. Update Versuch, nicht, das würde jetzt kommen. Eventuell habe ich dabei etwas "zerschossen".

Da es optional ist und ich eh nicht weiß, was ich damit machen sollte, also zumindest jetzt noch nicht, werde ich hier mit dem update vorsichtshalber mal aufhören.
